### ISSUE TITLE
Allow "$crate" to follow a `:vis`

### ISSUE THREAD

**Issue by CohenArthur** [Rust-GCC/gccrs#1061]
`:vis` fragments can be followed by "$crate", which we currently do not handle. We should look into whether or not it makes sense to add it as a reserved keyword (in rust-token.h, similarly to CRATE or SELF). 

Afterwards, we need to handle it properly in the `peculiar_fragment_match_compatible` function in `gcc/rust/parse/rust-parse.cc`. 

> FOLLOW(vis) = {`,` any keyword or identifier except a non-raw priv; any token that can begin a type; ident, ty, and path nonterminals}.
> ...
> The tokens that can begin a type are, as of this writing, {(, [, !, *, &, &&, ?, lifetimes, >, >>, ::, any non-keyword identifier, super, self, Self, extern, crate, __$crate__, _, for, impl, fn, unsafe, typeof, dyn}

Source: https://doc.rust-lang.org/reference/macro-ambiguity.html

**Comment by varun-r-mallya** [comment]
Can I work on this??

**Comment by CohenArthur** [comment]
Yes @varun-r-mallya, please do! I'll assign the issue to you :) If you have any questions or need guidance feel free to post your questions here

**Comment by varun-r-mallya** [comment]
Great! I am working on setting up the project itself right now. I'll ask further questions after I have successfully set it up. I might be a little slow for the next 2 weeks or so due to my exams, but I'll be pretty fast after that.
